### PR TITLE
feat(kevc): add known exploited vulnerability catalog 

### PR DESCRIPTION
**Pull request by masahiro331**
# Support KEVC 

https://www.cisa.gov/known-exploited-vulnerabilities-catalog

### CODE REVIEW

File: kevc/kevc_test.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+			name:      "sad path, invalid json",
+			inputFile: "testdata/sad/known_exploited_vulnerabilities.json",
+			wantErr:   "failed to KEVC json unmarshal",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				b, err := ioutil.ReadFile(tt.inputFile)
```
Comment (knqyf263) [inline kevc/kevc_test.go:35]: ioutil is deprecated

File: kevc/kevc.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+	kevcURL = "https://www.cisa.gov/sites/default/files/feeds/known_exploited_vulnerabilities.json"
+	retry   = 5
+	kevcDir = "kevc"
+)
+
+type Updater struct {
+	kevcURL     string
+	vulnListDir string
+	cacheDir    string
+	appFs       afero.Fs
```
Comment (knqyf263) [inline kevc/kevc.go:25]: Looks like it is not used. We can remove it.

File: kevc/kevc_test.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+		{
+			name:      "sad path, invalid json",
+			inputFile: "testdata/sad/known_exploited_vulnerabilities.json",
+			wantErr:   "failed to KEVC json unmarshal",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```
Comment (knqyf263) [inline kevc/kevc_test.go:34]: Can we use http.FileServer by renaming the file?
https://github.com/aquasecurity/vuln-list-update/blob/87765ef5608bfed13e7c73f5171b55f3017f0c21/go-vulndb/govulndb_test.go#L40

Or http.ServeFile.
https://github.com/aquasecurity/vuln-list-update/blob/3104324321c18b5962ebdf7284206333614d989f/alpine-unfixed/alpine_test.go#L60

File: kevc/kevc.go [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+	kevcURL = "https://www.cisa.gov/sites/default/files/feeds/known_exploited_vulnerabilities.json"
+	retry   = 5
+	kevcDir = "kevc"
+)
+
+type Updater struct {
+	kevcURL     string
+	vulnListDir string
+	retry       int
+}
```
Comment (knqyf263) [inline kevc/kevc.go:24]: I didn't see where it is used.

File: kevc/kevc.go [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+		if err := c.saveCVEPerYear(vuln); err != nil {
+			return xerrors.Errorf("failed to save KEVC per year: %w", err)
+		}
+		bar.Increment()
+	}
+	bar.Finish()
+	return nil
+}
+
+func (c Config) saveCVEPerYear(vuln Vulnerability) error {
```
Comment (masahiro331) [inline kevc/kevc.go:86]: We can use utils.SaveCVEPerYear.
But, the absolute path of "cacheDir" is added to the argument "dirName" in the function.
This causes an impact on the tmpDir used in the test case.

https://github.com/aquasecurity/vuln-list-update/blob/main/utils/utils.go#L35-L51

**Review by knqyf263 — APPROVED** [review]

**Pull request merged**